router/vendor.go

```diff
@@ -52,4 +54,44 @@ func NewVendorEngine(
 
 		ctx.JSON(http.StatusOK, res)
 	})
+
+	r.PUT(cfg.Put, func(ctx *gin.Context) {
+		id := ctx.Query("id")
+
+		body, err := io.ReadAll(ctx.Request.Body)
+		if err != nil {
+			ctx.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+		}
+
+		spec := PutVendorSpec{}
+		if err = json.Unmarshal(body, &spec); err != nil {
+			ctx.JSON(http.StatusInternalServerError, gin.H{
+				"error": err.Error(),
+			})
+		}
+
```

Comment:
checkout `ShouldBindJSON`, I think that will make our lives easier moving forward